src/Collection/ArrayCollection.php

```diff
@@ -0,0 +1,260 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Arrays\Collection;
+
+use ArrayAccess;
+use ArrayIterator;
+use Countable;
+use IteratorAggregate;
+use Yiisoft\Arrays\Collection\Modifier\ModifierInterface\AfterMergeModifierInterface;
+use Yiisoft\Arrays\Collection\Modifier\ModifierInterface\BeforeMergeModifierInterface;
+use Yiisoft\Arrays\Collection\Modifier\ModifierInterface\DataModifierInterface;
+use Yiisoft\Arrays\Collection\Modifier\ModifierInterface\ModifierInterface;
+
+final class ArrayCollection implements ArrayAccess, IteratorAggregate, Countable
+{
+    private array $data;
+
+    /**
+     * @var array|null Result array cache
+     */
+    private ?array $array = null;
+
+    /**
+     * @var ModifierInterface[]
+     */
+    private array $modifiers;
+
+    /**
+     * @param array $data
+     * @param ModifierInterface|ModifierInterface[] $modifiers
+     */
+    public function __construct(array $data = [], $modifiers = [])
+    {
+        $this->data = $data;
+        $this->modifiers = is_array($modifiers) ? $modifiers : [$modifiers];
```

Comment:
```suggestion
        $this->modifiers = (array) $modifiers;
```

Reply:
This not work. `$modifier` is object and will be converted to array.

Reply:
Ah, alright 👍